flink-runtime/src/main/java/org/apache/flink/runtime/highavailability/LeaderIdRegistry.java

```diff
@@ -15,11 +15,27 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+package org.apache.flink.runtime.highavailability;
 
-package org.apache.flink.runtime.resourcemanager;
+import java.util.UUID;
 
-import java.io.Serializable;
+/**
+ * Registry class to keep track of the current leader ID.
+ */
+public class LeaderIdRegistry {
```

Comment:
These are valid points, I will change the code to use the `LeaderRetrievalListener` instead.